mesonbuild/build.py

```diff
@@ -2066,20 +2065,18 @@ def process_kwargs(self, kwargs, environment):
                     self.vs_module_defs = File.from_absolute_file(path)
                 else:
                     self.vs_module_defs = File.from_source_file(environment.source_dir, self.subdir, path)
-                self.link_depends.append(self.vs_module_defs)
             elif isinstance(path, File):
                 # When passing a generated file.
                 self.vs_module_defs = path
-                self.link_depends.append(path)
             elif hasattr(path, 'get_filename'):
                 # When passing output of a Custom Target
-                path = File.from_built_file(path.subdir, path.get_filename())
-                self.vs_module_defs = path
-                self.link_depends.append(path)
+                self.vs_module_defs = File.from_built_file(path.subdir, path.get_filename())
             else:
                 raise InvalidArguments(
                     'Shared library vs_module_defs must be either a string, '
                     'a file object or a Custom Target')
+            self.process_link_depends(path, environment)
```

Comment:
I was wondering if this instead should be a shared method used by both the vs_module_defs and the link_depends kwargs handling.